modules/imgcodecs/test/test_png.cpp

```diff
@@ -7,6 +7,17 @@ namespace opencv_test { namespace {
 
 #if defined(HAVE_PNG) || defined(HAVE_SPNG)
 
+TEST(imRead, passing_Mat)
+{
+  Mat img = Mat::zeros(1000, 1000, CV_8UC3);
+  Mat subImg = img.rowRange(0, 512).colRange(0, 512);
```

Comment:
> `rowRange(0, 512)`

`rowRange(64, 512)`

> `colRange(0, 512);`

`colRange(128, 512);`

Reply:
sorry I don't understand what you are asking me to do here. Why do you want to copy it to another interval? this way it would then fail the test

Reply:
In that case test will check multiple strides as ROI will be an internal. Just a more strict test.

Reply:
ok right

Reply:
ok